### PR TITLE
Split up steps in Docker

### PR DESCRIPTION
**Pull request by solidsnack**
@ruimarinho @posix4e Would be great to get you guys's thoughts on this.

### CODE REVIEW

**Comment by posix4e** [comment]
I love this. I am too scared to see why the build popped though. On a different note, this will cause slightly more layers to our docker image. I think it's worth it.

**Comment by posix4e** [comment]
@solidsnack Maybe we should provide the reviewer with a bit of background with why we are doing this (docker as docs)

**Comment by ruimarinho** [comment]
Indeed :) I'm fine with this approach if you have a compelling use case of course.

**Comment by posix4e** [comment]
I think the trade-off we are making is that our dockerfile doubles as
documentation. I suppose we could combine both commands into one run
command to reduce image overhead

File: Dockerfile [hunk 1]
```diff
@@ -4,15 +4,12 @@ ENV PATH ~/.cargo/bin/:$PATH
...
-  && rm -rf /var/lib/apt/lists/* \
-  && curl -sf https://static.rust-lang.org/rustup.sh -o rustup.sh \
-  && bash rustup.sh --disable-sudo -y --verbose \
-  && pip install pgxnclient \
-  && cargo install rustfmt \
-  && mkdir -p "$CARGO_HOME"
+COPY util/docker-pg /tmp/docker-pg
+RUN /tmp/docker-pg
+
+COPY util/docker-rust /tmp/docker-rust
```
Comment (ruimarinho) [inline Dockerfile:10]: Group COPY instructions together?

```
COPY 

COPY

RUN 

RUN
```

File: util/docker-pg [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+
+export PATH=/usr/local/bin:"$PATH"
+
+pkgs=(ca-certificates curl git make gcc
+      postgresql-server-dev-"$PG_MAJOR"="$PG_VERSION"
+      python-pip)
+
+rm -rf /var/lib/apt/lists/*
```
Comment (ruimarinho) [inline util/docker-pg]: Should be done after apt-get install to remove unnecessary cache.

**Comment by ruimarinho** [comment]
Image size is not really important IMHO as the image is targeted for development, so if you think it makes it easier for new users I'm ok with it too.

**Comment by posix4e** [comment]
Should we merge or close

**Pull request closed without merging**